### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by maksym-nemera**
[DEMO LINK](https://maksym-nemera.github.io/react_pagination/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,18 +1,37 @@
...
-const items = getNumbers(1, 42)
-  .map(n => `Item ${n}`);
+import { Pagination } from './components/Pagination';
 
 export const App: React.FC = () => {
+  const totalItems = 42;
+  const firstPage = 1;
+
+  const [itemsOnPage, setItemsOnPage] = useState(5);
+  const [initialPage, setInitialPage] = useState(firstPage);
```
Comment (silchyn) [inline src/App.tsx:11]:
```suggestion
  const [page, setPage] = useState(firstPage);
```

File: src/components/Pagination/Pagination.tsx [hunk 2]
```diff
@@ -1 +1,103 @@
...
+}) => {
+  const [chosenPage, setChosenPage] = useState(currentPage);
+
+  useEffect(() => {
+    setChosenPage(currentPage);
+  }, [perPage]);
+
+  useEffect(() => {
+    onPageChange(chosenPage);
+  }, [chosenPage]);
```
Comment (silchyn) [inline src/components/Pagination/Pagination.tsx:26]: Why would you need that considering you have that state in the parent component?

**Review by silchyn — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 3]
```diff
@@ -34,62 +56,22 @@ export const App: React.FC = () => {
...
+        changedPage={changedPage}
+      />
+
       <ul>
-        <li data-cy="item">Item 1</li>
-        <li data-cy="item">Item 2</li>
-        <li data-cy="item">Item 3</li>
-        <li data-cy="item">Item 4</li>
-        <li data-cy="item">Item 5</li>
+        {getNumbers(firstPage, totalItems).map(item => (
```
Comment (dinaminahi) [inline src/App.tsx:67]: move this and slice into a variable

File: src/App.tsx [hunk 4]
```diff
@@ -1,18 +1,38 @@
...
+    : maxItems;
+
+  const selectedItems = (
+    event: ChangeEvent<HTMLSelectElement>,
+  ) => {
+    setItemsOnPage(+event.target.value);
+    setPage(firstPage);
+  };
+
+  const changedPage = (newPage: number) => {
```
Comment (dinaminahi) [inline src/App.tsx:26]:
```suggestion
  const changePage = (newPage: number) => {
```

File: src/App.tsx [hunk 5]
```diff
@@ -1,18 +1,38 @@
...
+  const [itemsOnPage, setItemsOnPage] = useState(5);
+  const [page, setPage] = useState(firstPage);
+
+  const maxItems = page * itemsOnPage;
+  const startVisibleItems = maxItems - itemsOnPage;
+  const endVisibleItems = maxItems > totalItems
+    ? totalItems
+    : maxItems;
+
+  const selectedItems = (
```
Comment (dinaminahi) [inline src/App.tsx:19]:
```suggestion
  const selectItemsPerPage = (
```

**Review by dinaminahi — CHANGES_REQUESTED** [review]
Great job 👍 
Let's do some fixes

**Review by 33Orange — APPROVED** [review]
Well done

**Pull request closed without merging**